### PR TITLE
Bug 1890130: fix pod creation deadlock

### PR DESCRIPTION
**Pull request by squeed**
Commit b5f89a66 introduced a deadlock on pod creation, but only for multitenant mode.

The fix is to lock runningPods as late as possible.

Fixes: 1890130

### CODE REVIEW

**Comment by squeed** [comment]
Single test failed; GCP issue:

` Failed to setup provider config for "gce": Error building GCE/GKE provider: unexpected response listing zones: googleapi: Error 503: The service is currently unavailable.`

/retest

**Comment by squeed** [comment]
More importantly, e2e-aws-multitenant is green (and we don't appear to have regressed networkpolicy).

File: pkg/network/node/pod.go [hunk 1]
```diff
@@ -288,6 +289,8 @@ func (m *podManager) processRequest(request *cniserver.PodRequest) *cniserver.Po
 		}
 		result.Err = err
 	case cniserver.CNI_DEL:
+		m.runningPodsLock.Lock()
+		defer m.runningPodsLock.Unlock()
```
Comment (danwinship) [inline pkg/network/node/pod.go:293]: for consistency should we drop the lock before calling `teardown`?

Comment (squeed) [reply pkg/network/node/pod.go:293]: definitely, good catch.

Comment (squeed) [reply pkg/network/node/pod.go]: fixed.

**Comment by squeed** [comment]
CI all green, just needs lgtm from someone.

**Pull request merged**